shell/platform/darwin/common/framework/Source/FlutterStandardCodec_Internal.h

```diff
@@ -16,22 +16,47 @@ typedef NS_ENUM(NSInteger, FlutterStandardField) {
   FlutterStandardFieldIntHex,
   FlutterStandardFieldFloat64,
   FlutterStandardFieldString,
-  // The following must match the corresponding order from `FlutterStandardDataType`.
   FlutterStandardFieldUInt8Data,
   FlutterStandardFieldInt32Data,
   FlutterStandardFieldInt64Data,
   FlutterStandardFieldFloat64Data,
   FlutterStandardFieldList,
   FlutterStandardFieldMap,
+  FlutterStandardFieldFloat32Data,
 };
 
 namespace flutter {
 FlutterStandardField FlutterStandardFieldForDataType(FlutterStandardDataType type) {
-  return (FlutterStandardField)(type + FlutterStandardFieldUInt8Data);
+  switch (type) {
+    case FlutterStandardDataTypeUInt8:
+      return FlutterStandardFieldUInt8Data;
+    case FlutterStandardDataTypeInt32:
+      return FlutterStandardFieldInt32Data;
+    case FlutterStandardDataTypeInt64:
+      return FlutterStandardFieldInt64Data;
+    case FlutterStandardDataTypeFloat32:
+      return FlutterStandardFieldFloat32Data;
+    case FlutterStandardDataTypeFloat64:
+      return FlutterStandardFieldFloat64Data;
+  }
 }
 FlutterStandardDataType FlutterStandardDataTypeForField(FlutterStandardField field) {
-  return (FlutterStandardDataType)(field - FlutterStandardFieldUInt8Data);
+  switch (field) {
+    case FlutterStandardFieldUInt8Data:
+      return FlutterStandardDataTypeUInt8;
+    case FlutterStandardFieldInt32Data:
+      return FlutterStandardDataTypeInt32;
+    case FlutterStandardFieldInt64Data:
+      return FlutterStandardDataTypeInt64;
+    case FlutterStandardFieldFloat32Data:
+      return FlutterStandardDataTypeFloat32;
+    case FlutterStandardFieldFloat64Data:
+      return FlutterStandardDataTypeFloat64;
+    default:
```

Comment:
Instead of using `default`, you can return the same default out of the switch statement. That way, unhandled enum values will be caught by the compiler if any additions are made.

Reply:
That praticular switch doesn't handle all cases for FlutterStandardField, only those that are mappable to FlutterStandardDataTypes. I'm  curious about the correct approach here if someone passes in a non "Data" field to the function. Assert?